### PR TITLE
feature: curtain support

### PR DESCRIPTION
**Pull request by daenny**
Initial curtain support. Needs some testing

### CODE REVIEW

**Comment by daenny** [comment]
Based on:
https://github.com/alextud/ha-switchbot-curtain/blob/main/custom_components/switchbot-curtain/cover.py

**Review by Danielhiversen — APPROVED** [review]
Awesome 👍 
Fix the failing style test and I will merge it

File: switchbot/__init__.py [hunk 1]
```diff
@@ -115,3 +123,24 @@ def turn_off(self) -> bool:
...
+    def close(self) -> bool:
+        """Send close command."""
+        return self._sendcommand(CLOSE_KEY, self._retry_count)
+
+    def stop(self) -> bool:
+        """Send stop command to device."""
+        return self._sendcommand(STOP_KEY, self._retry_count)
+
+    def set_position(self, position: int) -> bool:
+        """Send position command to device."""
```
[the remark this answers is not in the document]
Comment (daenny) [reply switchbot/__init__.py]: Done

**Comment by daenny** [comment]
I had to add a "pylint: disable=too-few-public-methods" for the base class. Is that OK for you?

**Pull request merged**